js/src/pages/settings/enhanced-conversions/useEnableEnhancedConversions.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY, useAppDispatch } from '~/data';
+
+/**
+ * Custom hook to manage the state of enhanced conversions.
+ * @return {Array} An array containing the current state and a function to toggle it.
+ */
+export const useEnableEnhancedConversions = () => {
+	const isEnabled = useSelect( ( select ) => {
+		const { getEnableEnhancedConversions } = select( STORE_KEY );
+		return getEnableEnhancedConversions();
+	} );
+
+	const { updateEnhancedConversionsStatus } = useAppDispatch();
+
+	const toggleEnhancedConversions = useCallback( async () => {
```

Comment:
@ankitrox As per my [comment here](https://github.com/woocommerce/google-listings-and-ads/pull/2948#discussion_r2128774884), let's have the selector return the state only and let's have the `updateEnhancedConversionsStatus` logic within the `SetupEnhancedConversions` component.